mailpoet/lib/Automation/Engine/Validation/WorkflowRules/AtLeastOneTriggerRule.php

```diff
@@ -0,0 +1,28 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Engine\Validation\WorkflowRules;
+
+use MailPoet\Automation\Engine\Data\Workflow;
+use MailPoet\Automation\Engine\Exceptions;
+use MailPoet\Automation\Engine\Validation\WorkflowGraph\WorkflowNode;
+use MailPoet\Automation\Engine\Validation\WorkflowGraph\WorkflowNodeVisitor;
+
+class AtLeastOneTriggerRule implements WorkflowNodeVisitor {
+  public const RULE_ID = 'at-least-one-trigger';
+
+  public function initialize(Workflow $workflow): void {
+    foreach ($workflow->getSteps() as $step) {
+      if ($step->getType() === 'trigger') {
+        return;
+      }
+    }
```

Comment:
You could also avoid this iteration by moving the check to `visitNode` and throwing the exception in `complete`. That said, I know in reality the performance difference will likely be unmeasurable...

Reply:
Addressed in a0ed13bd3